beacon-chain/sync/data_columns_reconstruct.go

```diff
@@ -3,210 +3,139 @@ package sync
-	if err := s.cfg.dataColumnStorage.Save(toSaveSidecars); err != nil {
-		return errors.Wrap(err, "save data column sidecars")
-	}
-
-	slotStartTime, err := slots.StartTime(s.cfg.clock.GenesisTime(), slot)
-	if err != nil {
-		return errors.Wrap(err, "failed to calculate slot start time")
-	}
-	log.WithFields(logrus.Fields{
-		"root":                          fmt.Sprintf("%#x", root),
-		"slot":                          slot,
-		"fromColumnsCount":              storedColumnsCount,
-		"sinceSlotStartTime":            time.Since(slotStartTime),
-		"reconstructionAndSaveDuration": time.Since(startTime),
-	}).Debug("Data columns reconstructed and saved")
-
-	// Update reconstruction metrics.
-	dataColumnReconstructionHistogram.Observe(float64(time.Since(startTime).Milliseconds()))
-	dataColumnReconstructionCounter.Add(float64(len(reconstructedSidecars) - len(verifiedSidecars)))
-
-	// Schedule the broadcast.
-	if err := s.scheduleMissingDataColumnSidecarsBroadcast(ctx, root, proposerIndex, slot); err != nil {
-		return errors.Wrap(err, "schedule reconstructed data columns broadcast")
-	}
-
-	return nil
-}
 
-// scheduleMissingDataColumnSidecarsBroadcast schedules the broadcast of missing
-// (aka. not seen via gossip but reconstructed) sidecars.
-func (s *Service) scheduleMissingDataColumnSidecarsBroadcast(
-	ctx context.Context,
-	root [fieldparams.RootLength]byte,
-	proposerIndex primitives.ValidatorIndex,
-	slot primitives.Slot,
-) error {
-	log := log.WithFields(logrus.Fields{
-		"root": fmt.Sprintf("%x", root),
-		"slot": slot,
-	})
-
-	// Get the time corresponding to the start of the slot.
-	genesisTime := s.cfg.clock.GenesisTime()
-	slotStartTime, err := slots.StartTime(genesisTime, slot)
-	if err != nil {
-		return errors.Wrap(err, "failed to calculate slot start time")
-	}
-
-	// Compute the waiting time. This could be negative. In such a case, broadcast immediately.
-	randFloat := s.reconstructionRandGen.Float64()
-	timeIntoSlot := broadcastMissingDataColumnsTimeIntoSlotMin + time.Duration(float64(broadcastMissingDataColumnsSlack)*randFloat)
-	broadcastTime := slotStartTime.Add(timeIntoSlot)
-	waitingTime := time.Until(broadcastTime)
-	time.AfterFunc(waitingTime, func() {
-		// Return early if the context was canceled during the waiting time.
-		if err := ctx.Err(); err != nil {
-			return
+		// Compute the slot start time.
+		slotStartTime, err := slots.StartTime(s.cfg.clock.GenesisTime(), slot)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to calculate slot start time")
 		}
 
-		if err := s.broadcastMissingDataColumnSidecars(slot, proposerIndex, root, timeIntoSlot); err != nil {
-			log.WithError(err).Error("Failed to broadcast missing data column sidecars")
-		}
-	})
+		// Randomly choose value before starting reconstruction.
+		randFloat := s.reconstructionRandGen.Float64()
+		timeIntoSlot := time.Duration(maxReconstructionDelaySec*randFloat) * time.Second
```

Comment:
won't this round to the nearest second? i'm wondering if this will do what is intended, the wait will either be 1 second or 2 second in this case right?